### PR TITLE
Transactional semantics for unique tokens + rollback for transfers affecting `nftsLedger`

### PR DESCRIPTION
**Pull request by joan41868**
Implemented fixes for:
#1919 - rollback should be called for nfts ledger upon failed transfer/
#1918 - implement transactional semantics for `UniqTokenViewsManager` - done by using 2 enums and keeping the pending changes in a map.

**Data structure used:**
Shot_20210809_170449

**Enums used:**
Shot_20210809_170028

**Transactional semantics implementation:**
Shot_20210809_170106

**Usage:**
Shot_20210809_170149

**Checklist:**
- [x] Tested - unit, e2e
- [x] Documented - javadoc

### CODE REVIEW

File: test-clients/src/main/java/com/hedera/services/bdd/spec/fees/FeesAndRatesProvider.java [hunk 1]
```diff
@@ -154,6 +154,7 @@ private void downloadFeeSchedule() throws Throwable {
 		byte[] bytes = response.getFileContents().getContents().toByteArray();
 		CurrentAndNextFeeSchedule wrapper = CurrentAndNextFeeSchedule.parseFrom(bytes);
 		feeSchedule = typePatching.withPatchedTypesIfNecessary(wrapper.getCurrentFeeSchedule());
+//		log.info("The patched fee schedule is {}", feeSchedule);
```
Comment (joan41868) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/fees/FeesAndRatesProvider.java:157]: will be reverted

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java [hunk 2]
```diff
@@ -860,8 +909,8 @@ public HapiApiSpec uniqueTokenTxnsAreAtomic() {
 				)
 				.when(
 						cryptoTransfer(
-								movingUnique(A_TOKEN, 1).between(TOKEN_TREASURY, SECOND_USER),
-								moving(101, B_TOKEN).between(TOKEN_TREASURY, FIRST_USER)
+								moving(101, B_TOKEN).between(TOKEN_TREASURY, FIRST_USER),
+								movingUnique(A_TOKEN, 1).between(TOKEN_TREASURY, SECOND_USER)
```
Comment (joan41868) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java:913]: will be reverted

File: hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java [hunk 3]
```diff
@@ -206,6 +217,9 @@ public void commit() {
 		if (nftsLedger != null && nftsLedger.isInTransaction()) {
 			nftsLedger.commit();
 		}
+		if (uniqTokenViewsManager != null && uniqTokenViewsManager.isInTransaction()) {
+			uniqTokenViewsManager.commit();;
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java:221]:
```suggestion
			uniqTokenViewsManager.commit();
```

**Review by Daniel-K-Ivanov — COMMENTED** [review]
Overall LGTM! Keep up the good work :)

**Pull request closed without merging**